src/datasources/email-api/pushwoosh-api.service.spec.ts

```diff
@@ -0,0 +1,129 @@
+import { FakeConfigurationService } from '@/config/__tests__/fake.configuration.service';
+import { PushwooshApi } from '@/datasources/email-api/pushwoosh-api.service';
+import { HttpErrorFactory } from '@/datasources/errors/http-error-factory';
+import { INetworkService } from '@/datasources/network/network.service.interface';
+import { CreateEmailMessageDto } from '@/domain/email/entities/create-email-message.dto.entity';
+import { DataSourceError } from '@/domain/errors/data-source.error';
+import { faker } from '@faker-js/faker';
+
+const networkService = {
+  post: jest.fn(),
+  delete: jest.fn(),
+} as unknown as INetworkService;
+const mockNetworkService = jest.mocked(networkService);
+
+const httpErrorFactory = {
+  from: jest.fn(),
+} as unknown as HttpErrorFactory;
+const mockHttpErrorFactory = jest.mocked(httpErrorFactory);
```

Comment:
I would avoid mocking our own classes at this level as by testing with the real one with are more confident on how the component behaves while deployed.

Reply:
I agree! Changed in [450b05c](https://github.com/safe-global/safe-client-gateway/pull/827/commits/450b05c653c741ecca3f181d55a27c308d405c12)